descwl_shear_sims/sim/sim.py

```diff
@@ -334,6 +341,342 @@ def make_seobs(
+    bands: list, optional
+        Default ['i']
+    epochs_per_band: int, optional
+        Number of epochs per band
+    noise_factor: float, optional
+        Factor by which to multiply the noise, default 1
+    cosmic_rays: bool
+        If True, add cosmic rays
+    bad_columns: bool
+        If True, add bad columns
+    """
+
+    se_dim = get_se_dim(coadd_dim=coadd_dim)
+
+    band_data = {}
+    for band in bands:
+
+        survey = get_survey(gal_type=galaxy_catalog.gal_type, band=band)
+        noise_per_epoch = survey.noise*np.sqrt(epochs_per_band)*noise_factor
+
+        # go down to coadd depth in this band, not dividing by sqrt(nbands)
+        # but we could if we want to be more conservative
+        mask_threshold = survey.noise*noise_factor
+
+        objlist, shifts = galaxy_catalog.get_objlist(
+            survey=survey,
+            g1=g1,
+            g2=g2,
+        )
+        objlist, shifts, bright_objlist, bright_shifts, bright_mags = get_objlist(
+            galaxy_catalog=galaxy_catalog,
+            survey=survey,
+            g1=g1,
+            g2=g2,
+            star_catalog=star_catalog,
+            noise=noise_per_epoch,
+        )
+
+        bdata_list = []
+        for epoch in range(epochs_per_band):
+            exp, noise_exp = make_exp(
+                rng=rng,
+                band=band,
+                noise=noise_per_epoch,
+                objlist=objlist,
+                shifts=shifts,
+                dim=se_dim,
+                psf=psf,
+                psf_dim=psf_dim,
+                dither=dither,
+                rotate=rotate,
+                bright_objlist=bright_objlist,
+                bright_shifts=bright_shifts,
+                bright_mags=bright_mags,
+                mask_threshold=mask_threshold,
+                cosmic_rays=cosmic_rays,
+                bad_columns=bad_columns,
+                star_bleeds=star_bleeds,
+            )
+            if galaxy_catalog.gal_type == 'wldeblend':
+                rescale_wldeblend_exp(
+                    survey=survey.descwl_survey,
+                    exp=exp,
+                    noise_exp=noise_exp,
+                )
+
+            # mark high pixels SAT and also set sat value in image for
+            # any pixels already marked SAT
+            saturate_image_and_mask(
+                image=exp.image.array,
+                bmask=exp.mask.array,
+                sat_val=BAND_SAT_VALS[band],
+            )
+            saturate_image_and_mask(
+                image=noise_exp.image.array,
+                bmask=noise_exp.mask.array,
+                sat_val=BAND_SAT_VALS[band],
+            )
+
+            bdata_list.append({'exp': exp, 'noise_exp': noise_exp})
+
+        band_data[band] = bdata_list
+
+    xoff = 3000
+    coadd_bbox = geom.Box2I(
+        geom.IntervalI(min=xoff + 0, max=xoff + coadd_dim-1),
+        geom.IntervalI(min=0, max=coadd_dim-1),
+    )
```

Comment:
Why is xoff fixed here?

Reply:
The point was just to have it not start at the zero of the larger reference grid.  I didn't think it mattered for this if we varied it.

If we want to get more realistic we can, making each slice part of a larger "tract" like lsst and having it tile that area.